svthalia/concrexit
# Improve promotion request deadlines

Closes #3529.



### Summary
- The default value for publish_date was removed, so that it needs to be set explicitly.
- The minimum timedelta was moved to each promotion channel.
  - This way it becomes possible to change this for individual channels (such as push notifs). Additionally, the default was set to 14 days as requested.
  - The current implementation does have one downside where changing it for all channels at once is not possible in the future, this will need to be done by hand for each channel.
- The PROMO_PUBLISH_DATE_TIMEDELTA setting was removed, as it couples unrelated things (the deadline and the email frequency) to each other. Also, changing it would break the weekly emails without changing their frequency, which is set elsewhere.

@WidadMajdoubi24 can you confirm this is the behaviour the Promocie would like? (The first two bullet points, the last one is all internal stuff.)


### How to test

1. Go to the admin page and create a promotion request
2. Check that it gets rejected or approved as appropriate, based on the channel and if the date was changed
3. Ideally, check if the emails are still correct (but they should behave the same as currently)
